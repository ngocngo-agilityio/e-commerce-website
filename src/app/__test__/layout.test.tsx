import { render } from '@testing-library/react';

// Components
import RootLayout from '../layout';

jest.mock('@providers', () => ({
  ...jest.requireActual('@providers'),
  ChakraUIProvider: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
}));

jest.mock('next-auth/react', () => ({
  ...jest.requireActual('next-auth/react'),
  SessionProvider: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
}));

describe('RootLayout', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <RootLayout>
        <div>Children component</div>
      </RootLayout>,
    );

    expect(container).toMatchSnapshot();
  });
});
