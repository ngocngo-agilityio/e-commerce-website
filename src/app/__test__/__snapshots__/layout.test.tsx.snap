// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout should match snapshot 1`] = `
<div>
  <html
    lang="en"
  >
    <body>
      <div>
        <div>
          <div>
            Children component
          </div>
        </div>
      </div>
    </body>
  </html>
</div>
`;
